src/python/tests/storage/test_storage_api.py

```diff
@@ -0,0 +1,283 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+import pytest
+import os
+import py
+from foglamp.storage.payload_builder import PayloadBuilder
+from foglamp.storage.storage import Storage
+
+__author__ = "Vaibhav Singhal"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+# TODO: remove once FOGL-510 is done
+@pytest.fixture(scope="module", autouse=True)
+def create_init_data(request):
+    """
+    Module level fixture that is called once for the test
+        Before the tests starts, it creates the init data
+        After all the tests, it clears database and sets the init data
+    Fixture called by default (autouse=True)
+    """
+    _dir = os.path.dirname(os.path.realpath(__file__))
+    file_path = py.path.local(_dir).join('/foglamp_test_storage_init.sql')
+    os.system("psql < {} > /dev/null 2>&1".format(file_path))
+    yield
+    os.system("psql < `locate foglamp_ddl.sql | grep 'FogLAMP/src/sql'` > /dev/null 2>&1")
+    os.system("psql < `locate foglamp_init_data.sql | grep 'FogLAMP/src/sql'` > /dev/null 2>&1")
```

Comment:
okay with init, but why ddl is needed?

Keeping the test data in mind; seems its not needed; just DELETE is enough with where condition.

Reply:
DDL is needed because init is not perfect at the moment. It does inserts without deleting the previous values for some tables. We may have duplicate rows if db is not cleaned. This all will go away with test environment setup and cleanup (FOGL-510)

Reply:
This is an area we will need to improve over time, it will only really become an issue when we have our second storage plugin and wish to support systems without Postgres. Ideally all DDL will be handled by the plugin and not by use of scripts. We will get rid of these scripts and replace them with a database agnostic mechanism.